resources/metadata/gctsRollback.yaml

```diff
@@ -0,0 +1,72 @@
+metadata:
+  name: gctsRollback
+  description: Perfoms roll back of one (default) or several commit(s)
+  longDescription: |
+    This step performs a rollback of commit(s) in a local ABAP system repository. If a <commit> parameter is specified, it will be used as the target commit for the rollback.
+    If no <commit> parameter is specified and the remote repository domain is 'github.com', the last commit with status 'success' will be used for the rollback. Otherwise,
+    gctsRollback will rollback to the previously active commit in the local repository.
+
+spec:
+  inputs:
+    secrets:
+      - name: abapCredentialsId
+        description: Jenkins credentials ID containing username and password for authentication to the ABAP system on which you want to perform the rollback
+        type: jenkins
+    params:
+      - name: username
+        type: string
+        description: User to authenticate to the ABAP system
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: password
+        type: string
+        description: Password to authenticate to the ABAP system
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+        secret: true
+      - name: repository
+        type: string
+        description: Specifies the name (ID) of the local repsitory on the ABAP system
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: host
+        type: string
+        description: Specifies the protocol and host adress, including the port. Please provide in the format '<protocol>://<host>:<port>'
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: client
+        type: string
+        description: Specifies the client of the ABAP system to be adressed
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: commit
+        type: string
+        description: Specifies the commit to deploy
+        scope:
+        - PARAMETERS
+        - STAGES
+        - STEPS
+      - name: githubPersonalAccessToken
```

Comment:
this should come in via a Jenkins credential.
We don't want people to maintain the token in the repo ... :-)